sklearn/utils/estimator_checks.py

```diff
@@ -3314,6 +3316,40 @@ def check_fit_idempotent(name, estimator_orig):
             )
 
 
+def check_fit_check_is_fitted(name, estimator_orig):
```

Comment:
We already have `check_estimators_unfitted` that make this check.
We could add some test for the pipeline and I am thinking as well to the SearchCV that are located in the `test_common.py`.

Reply:
That test doesn't check that the estimator actually passes `check_is_fitted` after fit, and doesn't pass it before fit though. You want me to add these tests there? To me it's a different test.